### PR TITLE
ENH: Name and type command line options of Windows module PowerShell script

### PR DESCRIPTION
**Pull request by SimonRit**
This PR aims at improving the management of arguments in `windows-download-cache-and-build-module-wheels.ps1` following problems encountered in  RTKConsortium/RTK#535, namely passing multiple cmake options on the command line. The difference in the behaviors of PowerShell and Python args calls for another scripting language in my opinion (see #250 discussion) but this PR is an improvement over #248 in terms of clarity and consistency with other platforms. It is not backward compatible with it but that is probably ok since it is a recent change. See https://github.com/RTKConsortium/RTK/actions/runs/4243933294/jobs/7377339692 and https://github.com/RTKConsortium/RTK/actions/runs/4243933347 for successful use.

### CODE REVIEW

**Comment by SimonRit** [comment]
(The last force-push is only a slight change of the doc.)

File: scripts/windows-download-cache-and-build-module-wheels.ps1 [hunk 1]
```diff
@@ -87,15 +99,14 @@ if (-not (Test-Path grep-win.zip)) {
...
-  }
-  else {
-    $build_command = "$build_command `"$($arg)`""
-  }
+$build_command = "& `"C:\Python$pythonVersion-x$pythonArch\python.exe`" `"C:\P\IPP\scripts\windows_build_module_wheels.py`" --no-cleanup --py-envs `"3$python_version_minor-x64`""
+if ("$setup_options".length -gt 0) {
+  $build_command = "$build_command $setup_options"
+}
+if("$cmake_options".length -gt 0) {
+  $build_command = "$build_command -- ""$cmake_options"""
```
Comment (tbirdso) [inline scripts/windows-download-cache-and-build-module-wheels.ps1:107]: Are extra quotes intended in this line? Probably the extra double quotes can be removed, or if they are required then they should be escaped with the ` character as done in line 102.

**Review by tbirdso — COMMENTED** [review]
Looking great, thank you for your work here @SimonRit ! One small comment on extra quotes, after that is resolved these changes LGTM.

File: scripts/windows-download-cache-and-build-module-wheels.ps1 (hunk 1 above)
Comment (SimonRit) [reply scripts/windows-download-cache-and-build-module-wheels.ps1:107]: Yes and they seem to be doing the same thing https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.3. For consistency, I'll change them to backslash quotes now, thanks for noticing.

Comment (tbirdso) [reply scripts/windows-download-cache-and-build-module-wheels.ps1]: You are right, thanks for the reference 👍

**Review by tbirdso — APPROVED** [review]

**Pull request merged**